tapestry.js

```diff
@@ -1212,32 +1213,35 @@ function updateViewedValue(id, amountViewedTime, duration) {
         
         // Save to database if logged in
         if (tapestryWpUserId) {
-            
-            if (id) {
-                var progData = {
-                    "post_id": tapestryWpPostId,
-                    "node_id": id,
-                    "progress_value": amountViewed
-                };
-                jQuery.post(TAPESTRY_PROGRESS_URL, progData, function(result) {})
-                .fail(function(e) {
-                    console.error("Error with adding progress data");
-                    console.error(e);
-                });
-            }
+            saveProgressCounter++;
+            // Send requests every 5 seconds
+            if (saveProgressCounter == TIME_BETWEEN_SAVE_PROGRESS) {
+                if (id) {
+                    var progData = {
+                        "post_id": tapestryWpPostId,
+                        "node_id": id,
+                        "progress_value": amountViewed
+                    };
+                    jQuery.post(TAPESTRY_PROGRESS_URL, progData, function(result) {})
+                    .fail(function(e) {
+                        console.error("Error with adding progress data");
+                        console.error(e);
+                    });
+                }
 
-            if (h5pVideoSettings && !isEmptyObject(h5pVideoSettings)) {
-                var h5pData = {
-                    "post_id": tapestryWpPostId,
-                    "json": JSON.stringify(h5pVideoSettings)
-                };
-                jQuery.post(TAPESTRY_H5P_SETTINGS_URL, h5pData, function(result) {})
-                .fail(function(e) {
-                    console.error("Error with adding h5p video settings");
-                    console.error(e);
-                });
+                if (h5pVideoSettings && !isEmptyObject(h5pVideoSettings)) {
+                    var h5pData = {
+                        "post_id": tapestryWpPostId,
+                        "json": JSON.stringify(h5pVideoSettings)
+                    };
+                    jQuery.post(TAPESTRY_H5P_SETTINGS_URL, h5pData, function(result) {})
+                    .fail(function(e) {
+                        console.error("Error with adding h5p video settings");
+                        console.error(e);
+                    });
+                }
+                saveProgressCounter = 0;
```

Comment:
You'll need to reset the timestamp here to the current time: `progressLastSaved = Date()`

Reply:
changed to use time stamp